### PR TITLE
[MRG+1] Improving docstring for symmetric_cbar argument and adding tests for _get_plot_stat_map_params

### PR DESCRIPTION
**Pull request by juhuntenburg**

### CODE REVIEW

File: nilearn/plotting/tests/test_img_plotting.py [hunk 1]
```diff
@@ -445,3 +446,171 @@ def test_plot_prob_atlas():
...
+    # data with positive and negative range
+    affine = np.eye(4)
+    data = np.array([[-.5, 1., 0.],
+                     [0., 1., -.2],
+                     [1.5, 2.5, 3.]])
+    img = nibabel.Nifti1Image(data, affine)
+
+    # symmtric_cbar set to True
+    cbar_vmin, cbar_vmax, vmin, vmax = _get_plot_stat_map_params(img, None,
+                                                                 True, {})
```
Comment (bthirion) [inline nilearn/plotting/tests/test_img_plotting.py]: _get_plot_stat_map_params(img,  vmax=None,  symmetric_cbar=True)  
would be more readable.

**Comment by bthirion** [comment]
Looks good otherwise

File: nilearn/plotting/tests/test_img_plotting.py [hunk 2]
```diff
@@ -445,3 +446,171 @@ def test_plot_prob_atlas():
...
+    assert_equal(vmin, -np.nanmax(data))
+    assert_equal(vmax, np.nanmax(data))
+    assert_equal(cbar_vmin, None)
+    assert_equal(cbar_vmax, None)
+    # same case if vmax has been set
+    cbar_vmin, cbar_vmax, vmin, vmax = _get_plot_stat_map_params(img, 2,
+                                                                 True, {})
```
Comment (GaelVaroquaux) [inline nilearn/plotting/tests/test_img_plotting.py]: Same remark as @bthirion 's here

**Comment by GaelVaroquaux** [comment]
LGTM. :+1: for merging.

Another review?

File: nilearn/plotting/tests/test_img_plotting.py [hunk 3]
```diff
@@ -445,3 +446,201 @@ def test_plot_prob_atlas():
                               threshold=0.2)
     # Testing the 4D plot prob atlas with contours
     display = plot_prob_atlas(img, view_type='continuous')
+
+
+def test_get_plot_stat_map_params_pos_neg():
+    # data with positive and negative range
+    affine = np.eye(4)
+    data = np.array([[-.5, 1., 0.],
```
Comment (lesteve) [inline nilearn/plotting/tests/test_img_plotting.py]: Given that the function is supposed to handle NaNs you could throw one or two NaNs in the matrix. The same applies for the other tests.

That could be done in a further PR too.

File: nilearn/plotting/img_plotting.py [hunk 4]
```diff
@@ -43,7 +43,10 @@
 def _get_plot_stat_map_params(stat_map_img, vmax, symmetric_cbar, kwargs,
     force_min_stat_map_value=None):
     """ Internal function for setting value limits for plot_stat_map and
-    plot_glass_brain
+    plot_glass_brain.
+    The limits for the colormap will always be set to range from -vmax to vmax.
+    The limits for the colorbar depend on the symmetric_cbar argument, please
+    refer to docstring of plot_stat_map).
```
Comment (lesteve) [inline nilearn/plotting/img_plotting.py]: There is a `)` without a `(` or am I being fooled by the diff.

Comment (juhuntenburg) [inline nilearn/plotting/img_plotting.py]: You're right, I fixed this.

File: nilearn/plotting/tests/test_img_plotting.py (hunk 3 above)
Comment (juhuntenburg) [inline nilearn/plotting/tests/test_img_plotting.py]: Included NaNs and reran the tests.

File: nilearn/plotting/img_plotting.py [hunk 5]
```diff
@@ -834,8 +837,9 @@ def plot_stat_map(stat_map_img, bg_img=MNI152TEMPLATE, cut_coords=None,
             symmetrical.
         symmetric_cbar: boolean or 'auto', optional, default 'auto'
             Specifies whether the colorbar should range from -vmax to vmax
-            or from 0 to vmax. Setting to 'auto' will select the latter if
-            the whole image is non-negative.
+            or from vmin to vmax. Setting to 'auto' will select the latter if
```
Comment (lesteve) [inline nilearn/plotting/img_plotting.py]: vmin is a tiny bit misleading given that it you can not pass vmin to these functions. I am not sure how to rephrase it better though, so we should probably leave that for later.

**Comment by GaelVaroquaux** [comment]
Tests are green. I am merging. This is a net improvement. Further improvement can be done later.

**Comment by lesteve** [comment]
> This is a net improvement. 

It is indeed, thanks a lot for this. It's great to see the colorbar code finally tested !

> Further improvement can be done later.

Agreed.

**Comment by lesteve** [comment]
Not sure how the landscape comments came back but I just disabled them. I seem to remember the consensus was that it was not worth the additional email noise.

You can look at the landscape.io website directly if you care about this kind of thing:
https://landscape.io/github/nilearn/nilearn

**Pull request merged**